### DISCUSSION TITLE
Power Average or Maximum not shown in Tour List

### DISCUSSION THREAD

**Discussion by HansVomSee** [mytourbook/mytourbook#1268] [General]
When open the "Analyse" View, you can see the average power and the max power of the tour. But in the Tour List these values are not shown.
It would be nice if these values are automatically updated if no vales are known during import process.

**Comment by FJBDev** [comment]
> But in the Tour List these values are not shown. It would be nice if these values are automatically updated if no vales are known during import process.

If by "the Tour List", you mean the "TourBook", those values are available

![image](https://github.com/mytourbook/mytourbook/assets/25235665/c34550b9-edab-42f2-8b3e-b9afc92f5bcd)

**Comment by HansVomSee** [comment]
Thanks for your response. But I have to explain it a more in detail:
In the analyse view I see the max power and average power, in the tour list I can turn on the colums for power max und power average - but the values are not shown.
For the Tour in the second row the values are shown because the device (a garmin 530) wrote it directly to the txc file. 
But for all other tours it would be nice if the values of the analyse view would be shown
![test](https://github.com/mytourbook/mytourbook/assets/39345673/c2f7761f-e5ad-4634-a5e6-ee41de6e6464)

**Reply by FJBDev** [reply]
> For the Tour in the second row the values are shown because the device (a garmin 530) wrote it directly to the txc file.

But even if the TCX file doesn't contain the average power values, MT will [compute it](https://github.com/mytourbook/mytourbook/blob/5fd00bfc1c9965a6108e3237a16b4c840dc8509d/bundles/net.tourbook.device.garmin/src/net/tourbook/device/garmin/GarminTCX_SAXHandler.java#L689)

>For the Tour in the second row the values are shown because the device (a garmin 530) wrote it directly to the txc file.
But for all other tours it would be nice if the values of the analyse view would be shown

Interesting... Which file format is the tour that doesn't show average/max power ? TCX ? FIT ?

**Reply by HansVomSee** [reply]
In my case this is a TCX Format (I export garmin activities as TCX Files)

**Comment by FJBDev** [comment]
> In my case this is a TCX Format (I export garmin activities as TCX Files)

Can you share a TCX file so I can see why the average power is  not computed ?

**Comment by HansVomSee** [comment]
Attached you find a sample file (where Watts data are available but without that MyTB calculates the average. 
[activity_13767365156.zip](https://github.com/mytourbook/mytourbook/files/14136639/activity_13767365156.zip)

**Reply by FJBDev** [reply]
Thanks for the file @HansVomSee , I will investigate and let you know

**Comment by FJBDev** [comment]
@HansVomSee After investigation, i [fixed ](https://github.com/mytourbook/mytourbook/pull/1282)this bug that should be available in the next version.
All you will have to do will be to "Reimport" the tours that are affected by this bug and you will see the max and average power populated for those tours

![image](https://github.com/mytourbook/mytourbook/assets/25235665/84559745-d60a-4b96-84c0-a0049b7b41ee)

![image](https://github.com/mytourbook/mytourbook/assets/25235665/4e7d663b-3265-44e0-9feb-f2a992edc9dd)

![image](https://github.com/mytourbook/mytourbook/assets/25235665/db63149e-d6fa-410e-aaff-682e7ff28980)

**Reply by HansVomSee** [reply]
Woah - thanks a lot. This is very nice - looking forward for the next version
